acct_mgt/moc_openshift.py

```diff
@@ -408,3 +408,27 @@ def create_limits(self, project_name, limits=None):
             "spec": {"limits": limits or self.get_limit_definitions()},
         }
         return api.create(body=payload, namespace=project_name).to_dict()
+
+    def get_users_in_project(self, project_name):
+        """
+        Returns a list of users that have a role in a given project/namespace
+        """
+        self.get_project(project_name)
+
+        users = []
+
+        api = self.get_resource_api(API_RBAC, "RoleBinding")
+
+        try:
+            role_binding_list = api.get(namespace=project_name)
+        except kexc.NotFoundError:
+            return users
+
+        for role_binding in role_binding_list.items:
+            subjects = role_binding.subjects
+
+            for subject in subjects:
+                if subject.kind == "User" and subject.name not in users:
+                    users.append(subject.name)
+
+        return users
```

Comment:
This should probably be restricted to the rolebindings we created in this code, which means those that match names in `OPENSHIFT_ROLES`.

If we use a `set` to hold the usernames we don't need to worry about checking for uniqueness.

That gets us something like:

```suggestion
        # Raise a NotFound error if the project doesn't exist
        self.get_project(project_name)

        users = set()
        role_binding_list = []

        for role in OPENSHIFT_ROLES:
            try:
                role_binding_list.append(self.get_rolebindings(project_name, role))
            except kexc.NotFoundError:
                continue

        for role_binding in role_binding_list:
            users.update(
                subject["name"]
                for subject in role_binding["subjects"]
                if subject["kind"] == "User"
            )

        return list(users)
```

Reply:
Also, this code needs some unit tests:

- Project does not exist
- Project exists but has no rolebindings
- Project has rolebindings for one role
- Project has rolebindings for multiple roles